script/signup.js

```diff
@@ -7,66 +8,80 @@ import {
   $pwVerifyAlertDiv,
   $form,
   $eyes,
-  mockUserInfo,
   message,
   drawAlert,
   eraseAlert,
   handlePwSight,
   validateEmailValue,
   validatePwValue,
   signupPageStatus,
+  url,
 } from './utils/auth.js';
 import { numberPattern as numPattern, englishPattern as engPattern } from './utils/regx.js';
 let { duplicateAccountState, pwFormState, comparePwState } = signupPageStatus;
 
-// 계정 중복 검사
-const checkDuplicateAccount = ({ value: emailValue }) => {
-  const validateResult = emailValue !== mockUserInfo.email;
-  validateResult === false /* 이메일이 mockUser와 같다면 */
-    ? (drawAlert($emailAlertDiv, $emailInput, message.duplicateEmail), (duplicateAccountState = false))
-    : (duplicateAccountState = true);
+if (localStorage.getItem('signUpAccessToken')) location.href = url.folderPage;
+
+const checkDuplicateAccount = async () => {
+  const { value: emailValue } = $emailInput;
+  if (!validateEmailValue()) {
+    return;
+  }
+  try {
+    await checkEmailAccess(emailValue);
+  } catch (error) {
+    if (!emailValue) {
+      drawAlert($emailAlertDiv, $emailInput, message.wrongEmail);
+      return;
+    }
+    drawAlert($emailAlertDiv, $emailInput, message.duplicateEmail);
+    duplicateAccountState = false;
+    console.error(error);
+    return;
+  }
+  duplicateAccountState = true;
 };
 
-// 비밀번호 양식 검사
 const checkPwForm = () => {
   const { value: pwValue } = $pwInput;
   const validateResult = engPattern.test(pwValue) && numPattern.test(pwValue) && pwValue.length >= 8;
-  validateResult === false /* 비밀번호 양식이 불일치하다면 */
+  validateResult === false
     ? (drawAlert($pwAlertDiv, $pwInput, message.wrongPwForm), (pwFormState = false))
     : (eraseAlert($pwAlertDiv, $pwInput), (pwFormState = true));
 };
 
-// 비밀번호 재입력 일치 확인
 const comparePw = () => {
-  const { value: pwValue } = $pwInput;
-  const { value: pwVerifyValue } = $pwVerifyInput;
-  if (!pwVerifyValue) {
+  if (!$pwVerifyInput.value) {
     drawAlert($pwVerifyAlertDiv, $pwVerifyInput, message.wrongPw);
   } else {
-    const validateResult = pwValue === pwVerifyValue;
-    validateResult === false /* 비밀번호와 비밀번호재입력 인풋값이 불일치하다면 */
+    const validateResult = $pwInput.value === $pwVerifyInput.value;
+    validateResult === false
       ? (drawAlert($pwVerifyAlertDiv, $pwVerifyInput, message.diffrentPw), (comparePwState = false))
       : (eraseAlert($pwVerifyAlertDiv, $pwVerifyInput), (comparePwState = true));
```

Comment:
validateResult 값이 boolean 이니 그 자체로 true, false 이므로 연산자를 이용하여 특정 값과 비교하지 않고 간결하게 작성이 가능합니다! 

```suggestions
validateResult
      ? (eraseAlert($pwVerifyAlertDiv, $pwVerifyInput), (comparePwState = true)) 
      : (drawAlert($pwVerifyAlertDiv, $pwVerifyInput, message.diffrentPw), (comparePwState = false));
```

참고로, 조건문 작성하실 때 조건은 true 일 때를 기준으로 먼저 작성하시는 것이 
else 일 때 이중 부정으로 뒤집어 생각하지 않아도 되어 편합니다!

Reply:
참고해보겠습니다!!